libusb/core.c

```diff
@@ -2185,15 +2188,12 @@ int API_EXPORTED libusb_set_option(libusb_context *ctx,
 		break;
 
 	/* Handle all backend-specific options here */
-	case LIBUSB_OPTION_USE_USBDK:
+	default:
```

Comment:
You need to add a `case` for `LIBUSB_OPTION_WEAK_AUTHORITY`, not move the default up.  The default is meant to catch invalid options.

Reply:
`usbi_backend.set_option` will catch  invalid options.

Reply:
No, that’s not the point or the design intent.  The core should validate the options.  The individual backends don’t need to validate, they just operate on what applies to them and return `LIBUSB_ERROR_NOT_SUPPORTED` otherwise.